### PR TITLE
Use v4 UUID generation

### LINKED ISSUE #2002: GUID: Experiencing very similar UUIDs being generated
@loganom 

Here's a list of some UUIDs being generated:
7D99B174-B2EC-11E5-B3F9-ACCD19563991
8398D366-B2EC-11E5-9070-ACCD19563991
87255784-B2EC-11E5-B102-ACCD19563991
CFB98150-B2EC-11E5-B92C-ACCD19563991
D4408D22-B2EC-11E5-8B30-ACCD19563991

They were created by searching for "uuid"

------
IA Page:  http://duck.co/ia/view/guid

### PR DESCRIPTION
**Pull request by gaulrobe**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [x] Improvement
    - [x] Bug fix: **`GUID: Fix #2002`**

###### Description of changes

GUID now uses v4 UUID creation. v4 UUIDs are random instead of being time-based and thus users are less likely to get similar UUID values as they use this IA.

###### Which issues (if any) does this fix?

Fixes #2002  - Leveraging the UUID::Tiny library to create v4 UUID strings

###### People to notify (@mention interested parties)

@moollaza @loganom @aleksen

---

Instant Answer Page: https://duck.co/ia/view/guid

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @loganom

### CODE REVIEW

**Comment by daxtheduck** [comment]
@gaulrobe Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please head to https://quackslack.herokuapp.com/ to get an invite.

>More Info: http://docs.duckduckhack.com/submitting/submitting-overview.html

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by mintsoft** [comment]
@moollaza FYI; this replaces `Data::GUID;` with `UUID::Tiny`

**Comment by mintsoft** [comment]
@loganom I've assigned myself to this for now, if you get a chance to look over this it'd be much obliged!

**Comment by mintsoft** [comment]
@moollaza This LGTM; do you know what would be required for the new dependecy?

**Comment by mintsoft** [comment]
@gaulrobe Would you be interested in converting this to use the Text Template at the same time here? We're requiring it for all new IA's; if not we can merge this in separately and I can convert it afterwards :+1:

**Comment by gaulrobe** [comment]
@mintsoft @moollaza Conversion to structured text template completed!

**Comment by loganom** [comment]
@mintsoft sorry for my absence - thanks for updating this IA @gaulrobe

**Comment by mintsoft** [comment]
@loganom No problem; real life often gets in the way :+1:

File: lib/DDG/Goodie/GUID.pm [hunk 1]
```diff
@@ -28,15 +28,20 @@ handle remainder => sub {
...
-    return $guid->as_string,
+    return $guid,
       structured_answer => {
-        input     => [],
-        operation => 'Random GUID',
-        result    => $guid->as_string
+          data => {
+            title => $guid,
+            subtitle => 'Random GUID',
+
```
Comment (mintsoft) [inline lib/DDG/Goodie/GUID.pm]: Tiny nit pick; can you bomb out the extra line here; also it should be formatted to 4 spaces/tab (many of these are randomly 2 spaces/tab here, I'm not sure why)

File: t/GUID.t [hunk 2]
```diff
@@ -2,25 +2,37 @@
...
     # Check that the trigger kicks in.
-    'guid'                       => test_zci(@answer),
-    'uuid'                       => test_zci(@answer),
-    'globally unique identifier' => test_zci(@answer),
-    'rfc 4122'                   => test_zci(@answer),
-    'new guid'                   => test_zci(@answer),
-    'random uuid'                => test_zci(@answer),
-    'generate new uuid'          => test_zci(@answer),
-    'generate random uuid'       => test_zci(@answer),
+    'guid'                       => test_zci(build_structured_answer()),
```
Comment (mintsoft) [inline t/GUID.t]: Can you switch this out to `build_test`; which is like:
```perl
sub build_test 
{
     return test_zci( ....... etc
}

**Comment by gaulrobe** [comment]
@mintsoft Thanks for the feedback! I've made those suggested changes 👍

**Comment by mintsoft** [comment]
Cool; this is just awaiting sign off around the new dependencies then

**Comment by GuiltyDolphin** [comment]
@mintsoft This has a *lot* of red labels! :smile:

**Comment by moollaza** [comment]
@gaulrobe @mintsoft UUID has been added to production 👍

**Comment by moollaza** [comment]
Deploying to beta...

**Comment by moollaza** [comment]
@gaulrobe Thanks! This LGTM :+1:

**Pull request merged**